### PR TITLE
SOC-1624 | Enable Discussions via WikiFactory

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-1624

### CODE REVIEW

File: front/scripts/main/components/community-badge.js [hunk 1]
```diff
@@ -47,10 +47,16 @@ export default App.CommunityBadgeComponent = Ember.Component.extend({
...
 
 		if (Ember.isEmpty(imageUrl)) {
-			return '';
+			// get wiki image
+			imageUrl = Ember.get(Mercury, 'wiki.image');
+		}
+
+		if (Ember.isEmpty(imageUrl)) {
+			// get default wiki image
+			imageUrl = '/front/images/community-badge-default.png';
```
Comment (rogatty) [inline front/scripts/main/components/community-badge.js]: Isn't brackets.svg enough?

File: front/scripts/main/components/community-badge.js [hunk 2]
```diff
@@ -47,10 +47,16 @@ export default App.CommunityBadgeComponent = Ember.Component.extend({
...
-		const imageUrl = this.get('badgeImages')[Ember.get(Mercury, 'wiki.id')];
+		let imageUrl = this.get('badgeImages')[Ember.get(Mercury, 'wiki.id')];
 
 		if (Ember.isEmpty(imageUrl)) {
-			return '';
+			// get wiki image
+			imageUrl = Ember.get(Mercury, 'wiki.image');
+		}
+
+		if (Ember.isEmpty(imageUrl)) {
```
Comment (rogatty) [inline front/scripts/main/components/community-badge.js]: This condition is duplicated.

File: server/facets/showApplication.js [hunk 3]
```diff
@@ -42,14 +42,16 @@ function getDistilledDiscussionsSplashPageConfig(hostName) {
...
  * @returns {void}
  */
-export default function showApplication(request, reply) {
+export default function showApplication(request, reply, wikiVariables) {
 	const wikiDomain = Utils.getCachedWikiDomainName(localSettings, request),
-		wikiVariables = new MW.WikiRequest({wikiDomain}).wikiVariables(),
 		context = {},
 		hostName = Utils.getWikiaSubdomain(request.info.host);
 
+	wikiVariables = typeof wikiVariables === 'undefined' || new MW.WikiRequest({wikiDomain}).wikiVariables();
```
Comment (rogatty) [inline server/facets/showApplication.js]: Looks nice but isn't easily readable, please simplify.

File: server/facets/showDiscussions.js [hunk 4]
```diff
@@ -1,53 +1,9 @@
-import config from '../../config/discussionsSplashPageConfig';
-import {getWikiaSubdomain} from '../lib/Utils';
+import * as MW from '../lib/MediaWiki';
+import * as Utils from '../lib/Utils';
```
Comment (rogatty) [inline server/facets/showDiscussions.js]: Please import only what's needed.

File: server/facets/showDiscussions.js [hunk 5]
```diff
@@ -1,53 +1,9 @@
...
-
-/**
- * Generates and shows splash page from config
- *
- * @param {Hapi.Request} request
- * @param {*} reply
- * @param {WikiaDiscussionsConfig} discussionsConfig
- * @returns {Hapi.Response}
- */
-function showSplashPage(request, reply, discussionsConfig) {
```
Comment (rogatty) [inline server/facets/showDiscussions.js]: If this is removed can we remove /config/discussionsSplashPageConfig too?

File: server/facets/showDiscussions.js [hunk 6]
```diff
@@ -1,53 +1,9 @@
-import config from '../../config/discussionsSplashPageConfig';
```
Comment (rogatty) [inline server/facets/showDiscussions.js]: Can we remove this file?

File: front/scripts/main/components/community-badge.js (hunk 2 above)
Comment (kvas-damian) [inline front/scripts/main/components/community-badge.js]: it's for purpose. When there is no wiki image fall back to default.

File: server/facets/showDiscussions.js (hunk 6 above)
Comment (kvas-damian) [inline server/facets/showDiscussions.js]: Not yet, it's used in https://github.com/Wikia/mercury/blob/5a92a3480ad579b30dcae6032d97421b7d9b8c6a/front/scripts/main/components/discussion-app-promotion.js#L22-21

File: front/scripts/main/components/community-badge.js (hunk 2 above)
Comment (rogatty) [inline front/scripts/main/components/community-badge.js]: You can use
```
imageUrl = Ember.getWithDefault(Mercury, 'wiki.image', '/front/images/community-badge-default.png');
```
to achieve the same.

File: server/facets/showDiscussions.js (hunk 6 above)
Comment (rogatty) [inline server/facets/showDiscussions.js]: Can we clean it up then? This component uses only two fields.

Comment (kvas-damian) [inline server/facets/showDiscussions.js]: There is a ticket for that https://wikia-inc.atlassian.net/browse/SOC-1575

File: front/scripts/main/components/community-badge.js (hunk 2 above)
Comment (kvas-damian) [inline front/scripts/main/components/community-badge.js]: Thanks for hint - forgot about it.

File: server/facets/showDiscussions.js (hunk 6 above)
Comment (rogatty) [inline server/facets/showDiscussions.js]: Ok, thanks!

File: server/facets/showApplication.js (hunk 3 above)
Comment (bkoval) [inline server/facets/showApplication.js]: Where is this variable declared?

Comment (kvas-damian) [inline server/facets/showApplication.js]: it's function parameter.

**Comment by bkoval** [comment]
LGTM :santa:

**Comment by rogatty** [comment]
LGTM

**Comment by rafalkalinski** [comment]
LGTM

**Pull request merged**